rs/replicated_state/src/canister_state/queues/refunds.rs

```diff
@@ -0,0 +1,112 @@
+use ic_types::{CanisterId, Cycles};
+use ic_validate_eq::ValidateEq;
+use ic_validate_eq_derive::ValidateEq;
+use std::collections::{
+    BTreeMap, BTreeSet,
+    btree_map::Entry::{Occupied, Vacant},
+};
+
+#[cfg(test)]
+mod tests;
+
+/// A prioritized pool of refunds to canisters. Used for accumulating outbound
+/// refunds at the subnet level, before routing into streams.
+///
+/// Refunds are prioritized by amount (larger amounts first). Ties are broken by
+/// canister ID (smaller IDs first).
+#[derive(Debug, Clone, Default, PartialEq, Eq, ValidateEq)]
+pub struct RefundPool {
+    // Pool contents.
+    #[validate_eq(Ignore)]
+    refunds: BTreeMap<CanisterId, Cycles>,
+
+    /// Refund priority queue. Holds all refunds, ordered by amount.
+    ///
+    /// Canister IDs break ties, ensuring deterministic ordering.
+    priority_queue: BTreeSet<RefundPriority>,
```

Comment:
Does this mean a refund for a very small amount can sit in the queue indefinitely if refunds for larger amounts keep coming in?